src/com/jambit/Database.java

```diff
@@ -0,0 +1,121 @@
+  private Properties catalogProps = new Properties();
+
+  private Connection activeDatabaseConnection;
+
+  /** Connects to the database */
+  public void connect() throws IOException, MultipleConnectionException, SQLException {
+
+    catalogProps.load(new FileInputStream(PROPERTY_PATH));
+    activeDatabaseConnection =
+        DriverManager.getConnection(
+            "jdbc:mysql://"
+                + catalogProps.getProperty("database.hostIP")
+                + ":"
+                + catalogProps.getProperty("database.port")
+                + "/"
+                + catalogProps.getProperty("database.databaseName"),
+            catalogProps.getProperty("database.username"),
+            catalogProps.getProperty("database.password"));
+
+    throw new MultipleConnectionException("Limited to one connection");
+  }
+
+  /**
+   * This function returns mood entries. When t = -1 this function returns all entries.
+   *
+   * @param t
+   * @return
+   * @throws SQLException
+   */
+  public ArrayList fetchMoodEntries(float t) throws SQLException {
+    long currentTime = getCurrentTime();
+    ArrayList<Integer> entries = new ArrayList<>();
+    StringBuilder query = new StringBuilder();
+    query
+        .append("SELECT * FROM ")
+        .append(catalogProps.getProperty("table.moodMeter"))
+        .append(" WHERE ")
+        .append(catalogProps.getProperty("table.moodMeter.time"));
+
+    if (t != -1) {
+      query.append(" BETWEEN ").append(currentTime - (t * 120)).append(" AND ").append(currentTime);
+    }
+
+    Statement st = activeDatabaseConnection.createStatement();
+    ResultSet rs = st.executeQuery(query.toString());
+    while (rs.next()) {
+      entries.add(rs.getInt(catalogProps.getProperty("table.moodMeter.entry")));
+    }
+
+    return entries;
+  }
+
+  public void writeMoodEntries(String input) throws SQLException, TimeOutException {
+    checkValid();
+
+    StringBuilder query =
+        new StringBuilder()
+            .append("INSERT INTO ")
+            .append(catalogProps.getProperty("table.moodMeter"))
+            .append("(")
+            .append(catalogProps.getProperty("table.moodMeter.entry"))
+            .append(",")
+            .append(catalogProps.getProperty("table.moodMeter.time"))
+            .append(")")
+            .append(" VALUES (")
+            .append(input)
+            .append(",")
+            .append(getCurrentTime())
+            .append(")");
+
+    Statement st = activeDatabaseConnection.createStatement();
+    st.executeUpdate(query.toString());
+  }
+
+  /**
+   * Get the Unix timestamp in seconds
+   *
+   * @return Unix timestamp in seconds
+   */
+  private long getCurrentTime() {
+    return System.currentTimeMillis() / 1000L;
+  }
+
+  /**
+   * Checks if MySQL connection is valid
+   *
+   * @throws TimeOutException Throws exception on timeout
+   */
+  public void checkValid() throws SQLException, TimeOutException {
```

Comment:
Instead of that check just print a warning that the connection is lost and you try to reconnect. After this attempt failed, you can create an exception